packages/node/src/core/evm/authorization/authorization-application.ts

```diff
@@ -9,94 +9,69 @@ import {
   LogsData,
   RequestErrorCode,
   RequestStatus,
-  WalletDataByIndex,
 } from '../../../types';
 
-function mapApiCalls(
-  apiCalls: ClientRequest<ApiCall>[],
+function applyAuthorization(
+  apiCall: ClientRequest<ApiCall>,
   authorizationsByEndpointId: AuthorizationByEndpointId
-): LogsData<ClientRequest<ApiCall>[]> {
-  const logsWithApiCalls: LogsData<ClientRequest<ApiCall>>[] = apiCalls.map((apiCall) => {
-    // Don't overwrite any existing error codes or statuses
-    if (apiCall.errorCode || apiCall.status !== RequestStatus.Pending) {
-      return [[], apiCall];
-    }
-
-    // There should always be an endpointId at this point, but just in case, check again
-    // and drop the request if it is missing. If endpointId is missing, it means that the
-    // template was not loaded
-    if (!apiCall.endpointId) {
-      const log = logger.pend('ERROR', `No endpoint ID found for Request ID:${apiCall.id}`);
-      const updatedApiCall = {
-        ...apiCall,
-        status: RequestStatus.Blocked,
-        errorCode: RequestErrorCode.TemplateNotFound,
-      };
-      return [[log], updatedApiCall];
-    }
+): LogsData<ClientRequest<ApiCall>> {
+  // Don't overwrite any existing error codes or statuses
+  if (apiCall.errorCode || apiCall.status !== RequestStatus.Pending) {
+    return [[], apiCall];
+  }
 
-    const isRequestedAuthorized = get(authorizationsByEndpointId, [apiCall.endpointId, apiCall.clientAddress]);
-
-    // If we couldn't fetch the authorization status, block the request until the next run
-    if (isNil(isRequestedAuthorized)) {
-      const log = logger.pend('WARN', `Authorization not found for Request ID:${apiCall.id}`);
-      const updatedApiCall = {
-        ...apiCall,
-        status: RequestStatus.Blocked,
-        errorCode: RequestErrorCode.AuthorizationNotFound,
-      };
-      return [[log], updatedApiCall];
-    }
+  // There should always be an endpointId at this point, but just in case, check again
+  // and drop the request if it is missing. If endpointId is missing, it means that the
+  // template was not loaded
+  if (!apiCall.endpointId) {
```

Comment:
I'm not sure I follow this, sorry. The `providerId` is already being supplied here:

https://github.com/api3dao/airnode/blob/5e896e94719fd86a09a5401c72378aa917489887/packages/node/src/core/evm/authorization/authorization-fetching.ts#L35-L43

`providerId` in this case is currently retrieved from the `config.nodeSettings`, although I think the plan is to always derive this from the mnemonic(?)

In any case, I know that we're also setting this when initializing each API call request object here:

https://github.com/api3dao/airnode/blob/5e896e94719fd86a09a5401c72378aa917489887/packages/node/src/core/evm/requests/api-calls.ts#L48

I assumed that `parsedLog.args.providerId` would always be the same as the `providerId` from the node settings, so it could be ignored in most cases and the assumption made that the `providerId` can always be found? Why would it need to be checked here too?

Reply:
You're right, I don't know what I was thinking about